cpp/src/arrow/util/compression.cc

```diff
@@ -131,7 +131,7 @@ Result<std::unique_ptr<Codec>> Codec::Create(Compression::type codec_type,
       if (compression_level_set) {
         return Status::Invalid("LZ4 doesn't support setting a compression level.");
       }
-      codec = internal::MakeLz4RawCodec();
+      codec = internal::MakeLz4HadoopRawCodec();
```

Comment:
This is a pain but I think we need to add `Compression::LZ4_HADOOP`

Reply:
Does this have to do with the earlier discussion with @pitrou? 

And just to clarify, by adding `Compression::LZ4_HADOOP` the goal is to distinguish the two codec implementations at the enum level? So we'd need to determine whether an incoming parquet file was compressed using Lz4 or Hadoop Lz4 by checking for file version in the parquet file metadata?

Reply:
No. Per below we now have 3 codec implementations:

* LZ4 raw
* LZ4 raw with Hadoop prefix
* LZ4 frame

This change makes "LZ4 raw" inaccessible. When we see "LZ4" in the Parquet metadata we should always instantiate the LZ4_HADOOP codec which has the logic to infer raw vs. raw w/ hadoop prefix. Does that make sense?

Reply:
@wesm is right. Also, we probably want to add specific tests in `arrow/util/compression_test.cc`